examples/bq-email-exports/send_email_function/main.py

```diff
@@ -0,0 +1,119 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Cloud Function for sending the email with a signed URL link.
+Triggered after export job from BigQuery to GCS is complete.
+Uses https://github.com/sendgrid/sendgrid-python
+"""
+
+import base64
+import datetime
+import json
+import logging
+import os
+
+from google.auth import default, exceptions, iam
+from google.auth.transport import requests
+from google.cloud import storage
+from google.oauth2 import service_account
+from sendgrid import SendGridAPIClient
+from sendgrid.helpers.mail import Mail
+
+
+def main(event, context):
+    """Entrypoint for Cloud Function"""
+
+    data = base64.b64decode(event['data'])
+    log_entry = json.loads(data)
+    status = log_entry['severity']
+    if status == "ERROR":
+        code = log_entry['protoPayload']['status']['code']
+        message = log_entry['protoPayload']['status']['message']
+        job_id = log_entry['protoPayload']['serviceData']['jobCompletedEvent'][
+            'job']['jobName']['jobId']
+        logging.error(
+            RuntimeError(
+                f"Error in upstream export job with ID {job_id}. Code {code}: {message}"
+            ))
+    else:
+        destination_uris = log_entry['protoPayload']['serviceData'][
+            'jobCompletedEvent']['job']['jobConfiguration']['extract'][
+                'destinationUris']
+        if len(destination_uris) > 1:
+            logging.warning(
+                "Multiple GCS URIs found from BigQuery export. Only the first "
+                "file will be linked in the email.")
+
+        blob_path = destination_uris[0]
+        blob = storage.Blob.from_string(blob_path)
+        url = generate_signed_url(blob) if get_env(
+            'SIGNED_URL') == 'True' else get_auth_url(blob_path)
```

Comment:
Nitty nit: I like to use strtobool for this sort of thing as it accepts more possible user inputs and does the right thing. 

https://docs.python.org/3/distutils/apiref.html#distutils.util.strtobool